app/v2/services/JsonMessageParsingService.scala

```diff
@@ -59,7 +59,11 @@ class JsonMessageParsingServiceImpl @Inject() (implicit materializer: Materializ
         .map {
           mt =>
             val root = mt.replace("\"", "")
-            MessageType.messageTypesSentByDepartureTrader.find(_.rootNode == root) match {
+            MessageType.messageTypesSentByDepartureTrader
+              .filterNot(
+                messageType => messageType == MessageType.DeclarationData
+              )
```

Comment:
I'm not convinced this is the right place to do this. This service is concerned with getting the message type out, and should not care about whether a message type is on an allow list.

Additionally, how will this work with arrivals?

I think a better way of doing this is to supply a `Seq[MessageType]` to the function that contains the message types you're willing to accept. That way, this code is agnostic to arrival/departure types, moving that responsibility to the appropriate place (the controller probably).

Reply:
Additionally, we're using this filtering code more than once. I wonder if we're better off just splitting the `messageTypesSentByDepartureTrader` (and arrivals version) into two lists:

```scala
val updateMessageTypesSentByDepartureTrader: Seq[MessageType] = Seq(
    // departure message types
)

val messageTypesSentByDepartureTrader: Seq[MessageType] = DeclarationData :+ updateMessageTypesSentByDepartureTrader
```

that way, we're not using `filterNot` all the time.

Reply:
Additionally, how will this work with arrivals? **This should be handled as part of Json Arrival json implementation and Prashant is working on on that part.**

I think a better way of doing this is to supply a Seq[MessageType] to the function that contains the message types you're willing to accept. That way, this code is agnostic to arrival/departure types, moving that responsibility to the appropriate place (the controller probably).    **This I have already done for xml under XmlParsers**
Also I have changed the MessageType list as suggested above